src/components/sidebar/nodeView/StopForm.tsx

```diff
@@ -9,223 +13,272 @@ import * as s from './stopForm.scss';
-                        label='PAIKAN NIMI'
-                        disabled={isEditingDisabled}
-                        value={stop.placeNameFi}
-                        onChange={onChange('placeNameFi')}
-                    />
-                    <InputContainer
-                        label='PAIKAN NIMI RUOTSIKSI'
-                        disabled={isEditingDisabled}
-                        value={stop.placeNameSe}
-                        onChange={onChange('placeNameSe')}
-                    />
-                </div>
-                <div className={s.flexRow}>
-                    <InputContainer
-                        label='OSOITE'
-                        disabled={isEditingDisabled}
-                        value={stop.addressFi}
-                        onChange={onChange('addressFi')}
-                    />
-                    <InputContainer
-                        label='OSOITE RUOTSIKSI'
-                        disabled={isEditingDisabled}
-                        value={stop.addressSe}
-                        onChange={onChange('addressSe')}
-                    />
-                </div>
-                <div className={s.flexRow}>
-                    <InputContainer
-                        label='POSTINUMERO'
-                        disabled={isEditingDisabled}
-                        value={stop.postalNumber}
-                        onChange={onChange('postalNumber')}
-                    />
-                    <Dropdown
-                        onChange={onChange('municipality')}
-                        codeList={municipalityCodeList}
-                        selected={stop.municipality}
-                        disabled={isEditingDisabled}
-                        label='KUNTA'
-                    />
-                </div>
-                </div>
+interface IStopFormState {
+    isLoading: boolean;
+    invalidPropertiesMap: object;
+    isEditingDisabled: boolean;
+}
+
+@inject('dialogStore', 'nodeStore', 'mapStore', 'errorStore')
+@observer
+class StopForm extends ViewFormBase<IStopFormProps, IStopFormState> {
+    constructor(props: IStopFormProps) {
+        super(props);
+        this.state = {
+            isLoading: false,
+            invalidPropertiesMap: {},
+            isEditingDisabled: false,
+        };
+    }
+
+    componentDidMount() {
+        this._validateAllProperties();
+    }
+
+    componentDidUpdate(prevProps: IStopFormProps) {
+        if (prevProps.stop.nodeId !== this.props.stop.nodeId) {
+            this._validateAllProperties();
+        }
+    }
+
+    private _validateAllProperties = () => {
+        const node = this.props.nodeStore!.node;
+        if (!node) return;
+        const stop = node.stop;
+        this.validateAllProperties(stopValidationModel, stop);
+        const isStopFormValid = this.isFormValid();
+        this.props.nodeStore!.setIsFormValid(isStopFormValid);
+    }
+
+    private onStopPropertyChange = (property: string) => (value: any) => {
```

Comment:
constistent naming, why does _validateAllProperties start with _, but onStopPropertyChange not?

Reply:
There are two validateAllProperties functions, one in IViewModel, one here so thats why I added _

Not using _ at onStopPropertyChange because that is not our convention

Reply:
validateStop?